product-service/src/handlers/post-products.ts

```diff
@@ -29,15 +29,15 @@ const postProducts: APIGatewayProxyHandler = async event =>  {
     await client.connect();
     console.log('DB was successfully connected!');
 
-    await client.query('BEGIN');
-    console.log('Transaction began!');
-
-    const { count, title, description, price } = JSON.parse(event.body);
+    const { count, title, description, price } = JSON.parse(event.body || event.Records[0].body);
```

Comment:
If this is the handler that is writing to the database, then it should be also publishing notifications to SNS. 
Besides, there can be more than one record in the batch.
And you better separate this from an API handler, or it can become extremely messy.

Reply:
Updated